src/dict.mjs

```diff
@@ -792,12 +792,12 @@ function withoutCollision(root, key) {
 /**
  * @template K,V
  * @param {undefined | Node<K,V>} root
- * @param {(value:V,key:K)=>void} fn
- * @returns {void}
+ * @param {(value:V,key:K)=>boolean|void} fn
+ * @returns {boolean}
  */
 function forEach(root, fn) {
```

Comment:
Rather than calling it foreach this function could be renamed to `all`, since it's checking that a predicate holds for every item of a dict

Reply:
This is what iterators are for in JavaScript. We could use those then we can use the regular loop syntax